src/factory.js

```diff
@@ -179,6 +179,14 @@ module.exports = ({ Component, createElement, PropTypes }) => {
             state = initialState(props);
 
             keys(state).forEach(k => {
+              if (
+                computed !== undefined &&
+                Object.prototype.hasOwnProperty.call(computed, k)
```

Comment:
You can simplify:
- test directly again `stateDescriptors` so you don't have to handle `undefined` (because only computed have created entries in `stateDescriptors`)
- you can use `in` here because `stateDescriptors` is an object without a prototype, so no collision risks.